scopesim/server/gitdir.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python3
+"""
+2022-04-10 (KL)
+Code taken directly from https://github.com/sdushantha/gitdir
+Adapted for ScopeSim usage.
+Many thanks to the authors!
+"""
+
+import re
+import os
+import urllib.request
+import signal
+import argparse
+import json
+import sys
```

Comment:
`sys` is only imported to do `sys.exit()`, which I think should not be done in a library. The other suggestions replace all those exit calls with a raise.
```suggestion
```